src/cephadm/cephadm

```diff
@@ -5212,6 +5261,548 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmCache:
+    task_types = ['disks', 'daemons', 'host', 'http_server']
+
+    def __init__(self):
+        self.started_epoch_secs = time.time()
+        self.tasks = {
+            "daemons": "inactive",
+            "disks": "inactive",
+            "host": "inactive",
+            "http_server": "inactive",
+        }
+        self.errors = []
+        self.disks = {}
+        self.daemons = {}
+        self.host = {}
+        self.lock = RLock()
+    
+    @property 
+    def health(self):
+        return {
+            "started_epoch_secs": self.started_epoch_secs,
+            "tasks": self.tasks,
+            "errors": self.errors,
+        }
+
+    def to_json(self):
+        return {
+            "health": self.health,
+            "host": self.host,
+            "daemons": self.daemons,
+            "disks": self.disks,
+        }
+
+    def update_health(self, task_type, task_status, error_msg=None):
+        assert task_type in CephadmCache.task_types
+        with self.lock:
+            self.tasks[task_type] = task_status
+            if error_msg:
+                self.errors.append(error_msg)
+
+    def update_task(self, task_type, content):
+        assert task_type in CephadmCache.task_types
+        with self.lock:
+            setattr(self, task_type, content)
```

Comment:
Minor improvement (I don't know the length of these), but if you store them (additionally) [gzipped](https://docs.python.org/3.8/library/gzip.html#gzip.compress), you can save some extra ms on each request. With dashboard we found this meant a noticeable improvement. And it's simple in terms of HTTP (just adding a 'Content-Encoding: gzip').

If this cached data is not used by anyone else than the HTTP exporter, why not saving a few of cycles/ms extra and simply storing the data in `gzip.compress(json.dumps(data).encode('utf-8'))`. There's a lot of processing that can be done upfront instead of on demand.